packages/sign-client/src/controllers/engine.ts

```diff
@@ -572,6 +591,402 @@ export class Engine extends IEngine {
+    // subscribe to response events
+    this.events.once<"session_connect">(engineEvent("session_connect"), onSessionConnect);
+    this.events.once(engineEvent("session_request", id), onAuthenticate);
+
+    try {
+      // send both (main & fallback) requests
+      await Promise.all([
+        this.sendRequest({
+          topic: pairingTopic,
+          method: "wc_sessionAuthenticate",
+          params: request,
+          expiry: params.expiry,
+          throwOnFailedPublish: true,
+          clientRpcId: id,
+        }),
+        this.sendRequest({
+          topic: pairingTopic,
+          method: "wc_sessionPropose",
+          params: proposal,
+          expiry: ENGINE_RPC_OPTS.wc_sessionPropose.req.ttl,
+          throwOnFailedPublish: true,
+          clientRpcId: fallbackId,
+        }),
+      ]);
+    } catch (error) {
+      // cleanup listeners on failed publish
+      this.events.off(engineEvent("session_connect"), onSessionConnect);
+      this.events.off(engineEvent("session_request", id), onAuthenticate);
+      throw error;
+    }
+
+    await this.setProposal(fallbackId, { id: fallbackId, ...proposal });
+
+    await this.client.auth.requests.set(id, {
+      authPayload: request.authPayload,
+      requester: request.requester,
+      expiryTimestamp,
+      id,
+      pairingTopic,
+      verifyContext: {} as any,
+    });
+
+    return {
+      uri: connectionUri,
+      response: done,
+    } as EngineTypes.SessionAuthenticateResponsePromise;
+  };
+
+  public approveSessionAuthenticate: IEngine["approveSessionAuthenticate"] = async (
+    sessionAuthenticateResponseParams,
+  ) => {
+    this.isInitialized();
+
+    const { id, auths } = sessionAuthenticateResponseParams;
+
+    const pendingRequest = this.getPendingAuthRequest(id);
+
+    if (!pendingRequest) {
+      throw new Error(`Could not find pending auth request with id ${id}`);
+    }
+
+    const receiverPublicKey = pendingRequest.requester.publicKey;
+    const senderPublicKey = await this.client.core.crypto.generateKeyPair();
+    const responseTopic = hashKey(receiverPublicKey);
+
+    const encodeOpts = {
+      type: TYPE_1,
+      receiverPublicKey,
+      senderPublicKey,
+    };
+
+    const approvedMethods: string[] = [];
+    const approvedAccounts: string[] = [];
+    for (const cacao of auths) {
+      const isValid = await validateSignedCacao({ cacao, projectId: this.client.core.projectId });
+      if (!isValid) {
+        const invalidErr = getSdkError(
+          "SESSION_SETTLEMENT_FAILED",
+          "Signature verification failed",
+        );
+
+        await this.sendError({
+          id,
+          topic: responseTopic,
+          error: invalidErr,
+          encodeOpts,
+        });
+
+        throw new Error(invalidErr.message);
+      }
+
+      const { p: payload } = cacao;
+      const recap = getRecapFromResources(payload.resources);
+
+      const approvedChains: string[] = [getNamespacedDidChainId(payload.iss) as string];
```

Comment:
`getNamespacedDidChainId` could potentially return `undefined`
```suggestion
      const namespacedDidChainId =  getNamespacedDidChainId(payload.iss)
      const approvedChains: string[] = namespacedDidChainId ? [namespacedDidChainId] : [];
```

Reply:
Decided to intentially omit this check as `payload.iss` is already validated during signature validations